### PR TITLE
Try: Allow space between on page list.

### LINKED ISSUE #29767: Page List inside Navigation block does not respect justification selection
## Description

When creating a menu with the Navigation block and the Page List block, the navigation alignment choice is not passed through to the Page List.

## Step-by-step reproduction instructions

Add a Navigation block.
Toggle the "Justification" icon, select a justification option e.g. 'Justify items right'
In the navigation block, add the Page List block to build a menu.

## Expected behaviour

The menu should respect the justification selection.

## Actual behaviour

The menu built from the Page List does not respect the justification of the parent Navigation block.

## Code snippet (optional)

With the extra UL of the page list, the justification is only applied to the first UL of the navigation block
```
<nav class="items-justified-right wp-block-navigation">
  <ul class="wp-block-navigation__container">
    <ul class="wp-block-page-list">
```
```
 .items-justified-right > ul {
    justify-content: flex-end;
}
```

## WordPress information
- WordPress version: 5.7
- Gutenberg version: 10.1.1
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? TT1-Blocks

## Device information
- Device: Desktop
- Operating system: Windows 10
- Browser: Firefox & Chrome

### PR DESCRIPTION
**Pull request by jasmussen**
## Description

Fixes #29767. 

The page list block, by virtue of being an additional container, did not work with the "space between" justification. 

This PR improves the situation in the editor, though the preview isn't perfect when using a mix of page list and other items, and make it perfect on the frontend. 

Editor before:

editor before

Frontend before:

frontend before

Editor after:

editor after

Frontend after:

Screenshot 2021-03-11 at 13 11 44

## How has this been tested?

Please test a page list block with some other menu items in a navigation block. Test all justifications.

Be sure to deselect the navigation block after choosing a justification.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by jasmussen** [comment]
"Display contents" has reasonable support and can be seen as a progressive enhancement for this particular edgecase: https://caniuse.com/css-display-contents

**Review by vcanales — COMMENTED** [review]
It works perfectly in the frontend, but in the editor I'm seeing the page items stack up vertically when I set justification to space between.

https://user-images.githubusercontent.com/1157901/110951335-b80ebe00-8323-11eb-8a9d-a9f8adadda9e.mov

The video is from Firefox, but I see the same behavior using Chrome.

Not sure how helpful this might be, but in the editor, after removing the `display: contents` rule, the component starts doing this dance:


https://user-images.githubusercontent.com/1157901/110952806-65ce9c80-8325-11eb-9728-c43688002d5a.mov

I tested T19 and TT1.

File: packages/block-library/src/page-list/editor.scss [hunk 1]
```diff
@@ -4,13 +4,29 @@
...
 		}
 	}
+
+	// space-between justification.
+	&.items-justified-space-between {
+		.wp-block-page-list > div,
+		.wp-block-page-list {
+			flex: 1;
+		}
+
```
Comment (vcanales) [inline packages/block-library/src/page-list/editor.scss]: I added the following and now it works as I think it was expected to:

```suggestion
		.wp-block-page-list > div,
		.wp-block-page-list {
			display: contents;
			flex: 1;
		}

```

https://user-images.githubusercontent.com/1157901/111011869-9b04da00-8379-11eb-933b-c8d4c456e51a.mov

**Comment by jasmussen** [comment]
That does work better, Vicente.

However I'm not super comfortable with the use of `display: contents;` — I think it's technically the right thing to do, but I also think this PR need a lot of testing and opinions. I'm going to let it sit for a bit to gather thought.

**Comment by scruffian** [comment]
IMO this is an improvement, but we should also look into removing the margins when space-between is selected so that they alignment is even:
Screenshot 2021-03-26 at 16 25 39

**Comment by jasmussen** [comment]
Thank you @scruffian, I'm happy to try this, even though the `display: contents;` is perhaps not ideal — it shouldn't break anything.

> we should also look into removing the margins when space-between is selected so that they alignment is even:

Agreed. I pushed a fix for that in https://github.com/WordPress/gutenberg/pull/30287, which touches other margins and therefore felt a better place to me:

Screenshot 2021-03-29 at 09 14 21

**Comment by jasmussen** [comment]
Is there still value in landing this one?

**Comment by scruffian** [comment]
Yes, it looks ready to land to me

**Comment by gwwar** [comment]
Checking in on this one @vcanales @scruffian @jasmussen, was this one still needed / what else needs to be done for landing? Another review?

**Comment by jasmussen** [comment]
Thanks for the check-in, Kerry. After Ben's last sanity check, I think this is ready for a final review. My only concern was and to an extent the use of `display: contents;` in https://github.com/WordPress/gutenberg/pull/29771/files#diff-842be739758d5e0052fe1a992ad4219200ee8e27fbbb8352f21fb592e31bd35bR37, however given it solves a edge-case problem I can see it as being a progressive enhancement.

**Review by getdave — APPROVED** [review]
✅  Tested this and works as described on both editor and frontend.

https://user-images.githubusercontent.com/444434/118459425-a961d000-b6f3-11eb-95b3-022d7aa4985a.mp4

**Comment by jasmussen** [comment]
Thanks so much 💫

**Review by carolinan — APPROVED** [review]
Working well for me in Chrome on Windows 10.

**Comment by jasmussen** [comment]
Thanks all!

**Pull request merged**